src/app.js

```diff
@@ -0,0 +1,275 @@
+    } else {
+      asset = `<asset id="${assetIDAttr}" name="${assetNameAttr}" start="0/1s" duration="${assetDurationAttr}" hasAudio="1" audioSources="1"><media-rep kind="original-media" src="${assetNameAttr}"/></asset>`
+      clip = `<asset-clip ref="${assetIDAttr}" name="${assetNameAttr}" start="${clipStartAttr}" duration="${clipDurationAttr}" offset="${clipOffsetAttr}" lane="2" enabled="1"/>`
+    }
+    this.#assetList.push(asset)
+    this.#clipList.push(clip)
+    this.#currentAssetID++
+    this.#currentOffset += clipDuration
+  }
+  stringify () {
+    let sequenceTCStartAttr = this.toTimecodeStr(this.#initialOffset)
+    let sequenceDurationAttr = this.toTimecodeStr(
+      this.#currentOffset - this.#initialOffset
+    )
+    if (this.#mediaInfo.hasVideo) {
+      this.#sequenceStr = `<sequence format="r0" tcFormat="NDF" tcStart="${sequenceTCStartAttr}" duration="${sequenceDurationAttr}"><spine>`
+    } else {
+      this.#sequenceStr = `<sequence format="r0" tcFormat="NDF" tcStart="${sequenceTCStartAttr}" duration="${sequenceDurationAttr}"><spine><gap name="Gap" start="${sequenceTCStartAttr}" duration="${sequenceDurationAttr}" offset="${sequenceTCStartAttr}">`
+    }
+    return (
+      this.#startStr +
+      this.#assetList.join('') +
+      this.#eventStr +
+      this.#sequenceStr +
+      this.#clipList.join('') +
+      this.#endStr
+    )
+  }
+} // class FCPXML
+
+let ffmpeg
+
+// Check browser support and load libraries
+async function initialize () {
+  if (typeof SharedArrayBuffer === 'undefined') {
+    document.getElementById('message').innerHTML =
+      'Error: Please use latest Chrome/Firefox/Edge'
+  }
+  ffmpeg = FFmpeg.createFFmpeg({ log: true })
+  await ffmpeg.load()
+}
+
+function download (url, fileName) {
+  let link = document.createElement('a')
+  link.href = url
+  link.download = fileName
+  document.body.appendChild(link)
+  link.click()
+  link.remove()
+}
+
+// Convert editList to a proper output format
+function generateOutput (mediaInfo, editList, format) {
+  let output
+  if (format === 'fcpxml') {
+    output = new FCPXML(mediaInfo)
+  } // add formats here
+  for (let i = 0; i < editList.length; i += 2) {
+    output.addClip(editList[i], editList[i + 1])
+  }
+  let blob = new Blob([output.stringify()], { type: 'text/plain' })
+
+  return window.URL.createObjectURL(blob)
+}
+
+async function edit (mediaInfo) {
+  let editList = [] // [clipStart, clipEnd, ... ]: clip means segment to keep
+  let silenceStartMsgs = []
+  let silenceEndMsgs = []
+  ffmpeg.setLogger(({ message }) => {
+    if (message.includes('silence_start')) {
+      silenceStartMsgs.push(message)
+    } else if (message.includes('silence_end')) {
+      silenceEndMsgs.push(message)
+    }
+  }) // start logging after this line
+  await ffmpeg.run(
+    '-i',
+    mediaInfo.fileName,
+    '-af',
+    'silencedetect=n=-27dB:d=1',
```

Comment:
feel n and d can both be separate variable instead of hard-wired in the string.

Reply:
Yep, that will be done when we implement threshold detection and user-determined duration, for now though it's just hard coded